fetch.bs

```diff
@@ -3898,20 +4170,32 @@ these steps:
 
  <li><p>Let <var>actualResponse</var> be null.
 
+ <li><p>Let <var>timingInfo</var> be <var>fetchParams</var>'s <a for="fetch params">timing info</a>
+
  <li>
   <p>If <var>request</var>'s <a>service-workers mode</a> is "<code>all</code>", then:
 
   <ol>
    <li><p>Let <var>requestForServiceWorker</var> be a <a for=request>clone</a> of
    <var>request</var>.
 
+   <li><p>Set <var>fetchParams</var>'s <a for="fetch params">timing info</a>'s
+   <a for="fetch timing info">worker start time</a> to the <a for=/>unsafe shared current time</a>.
+
    <li><p>Set <var>response</var> to the result of invoking <a for=/>handle fetch</a> for
    <var>requestForServiceWorker</var>. [[!HTML]] [[!SW]]
 
    <li>
     <p>If <var>response</var> is not null, then:
 
     <ol>
+     <li><p>Set <var>fetchParams</var>'s <a for="fetch params">timing info</a>'s
+     <a for="fetch timing info">response start time</a> to the <a for=/>unsafe shared current time</a>.
+
+     <li><p>Set <var>timingInfo</var>'s <a for="fetch timing info">encoded body size</a> and
+     <a for="fetch timing info">decoded body size</a> to <var>response</var>'s
+     `<code>Content-Length</code>` <a for=/>header</a>.
```

Comment:
You want https://fetch.spec.whatwg.org/#header-list-extract-a-length here, right? (Similar elsewhere.)

Should we also set this field for service worker responses? Some tests would be good here.

Reply:
I don't understand the proposal, this line specifically sets this field for service-worker response. It's roughly tests, I agree that more tests would be better.

Reply:
The problem is what it sets the field to. "response's Content-Length header" is not a thing. A response has a header list and from there you need to get the header and then you need to parse it into a value. Extract a length does that for `Content-Length`.

Reply:
Yes I got that part, I meant about the comment: "Should we also set this field for service worker responses".